tests/diagnostics/uninit-var.slang

```diff
@@ -0,0 +1,35 @@
+// uninit-var.slang
+//TEST:SIMPLE(filecheck=CHECK): -target hlsl -entry computeMain -profile cs_6_0
+
+//DISABLED_TEST(compute):COMPARE_COMPUTE_EX:-slang -compute -shaderobj
+//DISABLED_TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute -shaderobj
+//DISABLED_TEST(compute):COMPARE_COMPUTE_EX:-d3d12 -compute -shaderobj
+
+struct Data
+{
+    float3 value;
+};
+
+RWStructuredBuffer<Data>  inputBuffer;
+RWStructuredBuffer<Data> outputBuffer;
+
+float foo(out float3 v)
+{
+    float uninitializedVariable;
+
+    // CHECK: warning
```

Comment:
This check will always pass and you need to make the check more specific to the warning you are expecting.
I think you can make it something like
```
// CHECK: warning 4101{{[56]}}
````

The reason why it will still pass when there was no warning about the uninitialized variable is that the output shader without any warning or error will have the following line.
```
#ifndef __DXC_VERSION_MAJOR
    // warning X3557: loop doesn't seem to do anything, forcing loop to unroll
    #pragma warning(disable: 3557)
#endif
```

I also noticed that the warning you are getting is not about the intended variable, "uninitializedVariable" and it is about "v" not being set for the other components "y" and "z".
```
tests/temp2.slang(22): error 41015: use of uninitialized value 'v'
    float r = v.x + uninitializedVariable;
                ^
tests/temp2.slang(23): warning 41016: returning without initializing out parameter 'v'
    return r;
    ^~~~~~
}
```
If you remove the keyword "out" from "float foo(out float3 v)", the warning doesn't show up.
In other words, I don't think we are printing warnings for the uninitialized variables.

Reply:
thanks! I updated the test to add explicit checks. now I have a real issue to deal with.